### ISSUE TITLE
How to transfer feature maps

### ISSUE THREAD

**Issue by gusdlgustn** [kjy7567/speech_emotion_recognition_from_log_Mel_spectrogram_using_vertically_long_patch#1]
Hello. How to match the feature maps of the student network and the trained teacher network? I trained teacher model and save it. I'd appreciate your help.

**Comment by kjy7567** [comment]
Hello.
The teacher network and student network have the output y before passing through the MLP classifier.
In model.py
```python
class Teacher(nn.Module):
...
    def forward(self, img):
    ...
        return x, y

class Student(nn.Module):
...
    def forward(self, img):
    ...
        return x, y
```

If you have already trained the teacher network, you should freeze the weights of the teacher network and train y using the mean absolute error (L1 loss).

You can freeze teacher network by using below 
```python
teacher.eval()
...
with torch.no_grad():
    x1, y1 = teacher(img)
...
x2, y2 = student(img)
...
```
or
```python
teacher.eval()
...
for para in teacher.parameters():
    para.requires_grad = False
...
x1, y1 = teacher(img)
x2, y2 = student(img)
...
```
And then calculate mean absolute error (MAE)
```python
x1, y1 = teacher(img) # fixed weight
x2, y2 = student(img)

mae = MAE(y1, y2) # You may need to modify y1.requires_grad_(True) because of fixed teacher weights.
ce = CE(x2, label) # Cross entropy loss
loss = ce + 10 * mae
loss.backward()
```

**Comment by gusdlgustn** [comment]
Is the part you explained "iii: (c) Train the student network with the feature map matching performed using cross entropy loss + α * L1 loss."?
Should "(b) Match the feature maps of the student network and the trained teacher network using Mean absolute error(L1 loss)" come before the code you explained? I'm sorry to keep asking you questions. I think I'm asking the basics.

**Comment by kjy7567** [comment]
Feel free to ask me any questions you may have. :)
Sorry, I made a mistake.
Your question is corresponds to 'ii: (b) Match the feature maps of the student network and the trained teacher network using Mean absolute error(L1 loss).'
You should have matched the features using only MAE loss, but I used cross entropy incorrectly.

```python
_, y1 = teacher(img) # fixed weight
_, y2 = student(img)

loss = MAE(y1, y2) # You may need to modify y1.requires_grad_(True) because of fixed teacher weights.
loss.backward()
```

At this stage, you also need to freeze the weights of the teacher network.
After this feature matching stage, you need to train the student network using the cross entropy loss + α * L1 loss as explained above.

**Comment by gusdlgustn** [comment]
Thank you for your kind reply. Have a nice day!
